test/gate_test.dart

```diff
@@ -421,5 +432,19 @@ void main() {
           gtm.generateSynth(), gtm.runtimeType.toString(), vectors);
       expect(simResult, equals(true));
     });
+
+    test('Index by Logic test', () async {
+      var gtm = IndexGateTestModule(Logic(width: 8), Logic(width: 8));
+      await gtm.build();
+      var vectors = [
+        Vector({'a': 14, 'b': 0}, {'a_indexedby_b': 0}),
+        Vector({'a': 14, 'b': 2}, {'a_indexedby_b': 1})
```

Comment:
add some vectors to test invalid a and/or invalid b produce an invalid output (some other tests have examples of this)